src/main/java/io/vertx/ext/mongo/IndexOptionDefaults.java

```diff
@@ -0,0 +1,52 @@
+package io.vertx.ext.mongo;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+import java.util.Objects;
+
+@DataObject(generateConverter = true)
+public class IndexOptionDefaults {
+  private JsonObject storageEngine = new JsonObject();
+
+  public IndexOptionDefaults() {
+    storageEngine = new JsonObject();
```

Comment:
Is this the right name for the property? It looks like a copy/paste.

Also, why do we need a separate data object if the value is a single `JsonObject`?

Reply:
As described here https://docs.mongodb.com/manual/reference/method/db.createCollection/

```
db.createCollection( <name>,
   {
     capped: <boolean>,
     timeseries: {                  // Added in MongoDB 5.0
        timeField: <string>,        // required for time series collections
        metaField: <string>,
        granularity: <string>
     },
     expireAfterSeconds: <number>,
     autoIndexId: <boolean>,
     size: <number>,
     max: <number>,
     storageEngine: <document>,
     validator: <document>,
     validationLevel: <string>,
     validationAction: <string>,
     indexOptionDefaults: <document>,
     viewOn: <string>,              // Added in MongoDB 3.4
     pipeline: <pipeline>,          // Added in MongoDB 3.4
     collation: <document>,         // Added in MongoDB 3.4
     writeConcern: <document>
   }
)
```
`indexOptionDefaults` is part of the options we can pass to a `createCollection` call. Adding now all of those fields and document their respective intentions and semantics are imho out of scope for the change, described in #256. Which doesn't mean, that we can not later on add these options to be passed to the mongo driver. Having this `DataObject` already in place eases the possibility to extend it later on, when needed.

Reply:
Looking at the Mongo documentation, it seems `storageEngine` should be part of `CreateCollectionOptions`, not nested in `IndexOptionsDefaults`.

Reply:
You're right. I am a sloppy reader. Thanks for clarifying

Reply:
After reconsidering the documentation it seems that `CreateCollectionOptions` takes `storageEngine` options at the root level of the options. But `indexOptionDefaults` takes another `storageEngine` object, which is used as default for created indexes on that collection. 
As the mongodb driver implements it like this

 https://github.com/mongodb/mongo-java-driver/blob/84910ac76988cebcd2730db77c167a4d685efceb/driver-core/src/main/com/mongodb/client/model/CreateCollectionOptions.java#L38

and

https://github.com/mongodb/mongo-java-driver/blob/84910ac76988cebcd2730db77c167a4d685efceb/driver-core/src/main/com/mongodb/client/model/IndexOptionDefaults.java#L30

I thought implementing a corresponding dataobject would be the straight forward way.

@tsegismont I could merely take a `JsonObject` as `IndexOptionDefaults` and simply look for `storageEngine` key, but on the other hand, having a corresponding dataObject class might be a cleaner solution. I'll change it as you wish. Just let me know how you decide :)

Reply:
If storage engine is non structured data (it seems so reading the Mongo doc), then a mere `JsonObject` will do.